File: products/src/main/java/com/github/products/services/ICategoryService.java
package com.github.products.services;

import com.github.products.entity.Category;

import java.util.List;

public interface ICategoryService {

    List<Category> readAllStatusOn();

    Category create(Category c);

    Category readByName(String name);

    Category getByName(String name);

    Category readById(Long id);

    void update(Category c);

    void removeId(Long id);

    List<Category> readAll();

}

File: server-mocks/src/main/kotlin/com/github/server/mocks/services/ICategoryService.kt
package com.github.server.mocks.services

import com.github.server.mocks.payload.Category

interface ICategoryService {

    fun create(category: Category): Category

    fun readByName(name: String): Category

    fun readAll(): List<Category>

    fun update(category: Category)

    fun remove(id: Long)

}